### ISSUE TITLE
Visually distinguish mandatory, recommended and optional fields in the UI

### ISSUE THREAD

**Issue by ninsbl** [NINAnor/NOFAInsert#66]
In order to make it more transparent for the user what data she or he has to or should enter into the DB "mandatory", "recommended" and "optional" fields should be visually distinguished in the UI.
Here, 
 - bold font type,
 - (possibly differently colored) asterisk signs
 - outline color of input widgets or
 - grouping of widgets
could be used.

In order to get this done we need the following: 
 - [x] an overview over which fields are mandatory, and which are optional (This is probably a job for @andersfi ) 
 - [x] implementation of the former (likely something for @svoboond ; feel free to suggest a solution you find visually most appealing or most advantageous to implement)

Added this to milestone 0.7, feel free to change if you like!

**Comment by andersfi** [comment]
In the following I list mandatory fields by table (referring to db table here). Note that non-mandatory fields when occurring in drop-down lists MUST have NULL as default value. Otherwise, one would get a lot of erroneous input caused by user sloppiness. 
## event
- samplingProtocol
- dateEnd
- recordedBy
- reliability

Notes on event: fieldNumber should be added as input - this is a reffererence to the orginal identifier/paper form and can be very useful when debugging data. 

## occurrence
- taxon
- occurrenceStatus ('present' as default)
- establishmentMeans ('unknown' as default)
Notes on occurrence: recordNumber should be added as input - references the orginal identifier of the data. very useful in debugging of data. fyi: As for fieldNumber this should be unique within dataset (not implemented in DB, and not current priority). 

NOTE: ecotype must have NULL as default, when I set taxon:salmo trutta, Anadromous Brown Trout pops up as default - this is dangerous behaviour - and time consuming as I would manually have to reset the input to NULL (btw currently not possible either) for 95% of my inputs (which does not link to an ecotype). 

## m_dataset
- datasetName
- datasetID 
- ownerInstitutionCode
- rigthsHolder 
- accessRigths

## m_project
- organization
- projectd number
- startYear
- projectLeader
- financer

## m_reference
Note: refferences are tricky, and the required input does really depend upon default input really depends upon the referenceType. However, suggest not putting more effort into this at the moment. Lot of work for little gain. 
- title
- author(s)
- year 
- refferenceType

**Comment by svoboond** [comment]
Fields "recordNumber" "fieldNumber" have been added and default values of non-mandatory widgets have been set to None.

Widgets that needs to be selected/filled have red background. Once a widget contains valid data its color is set to default.

However, there is an issue with date edits because QDateEdit can not be empty. Currently date edits that are mandatory have yellow background. Once user clicks on mandatory date edit it is considered as selected and its color changes to default. This is not too good but I could not find a better solution. The same problem goes for non-mandatory date edits. If you have an idea how to solve it please tell me.

Other than that the issue is solved. I work on this in "mandatory_fields" branch and so far it has not been merged with master.

**Comment by svoboond** [comment]
Default values of "occurrenceStatus" and "establishmentMeans" have been set.

```
ALTER TABLE nofa.occurrence
    ALTER COLUMN "occurrenceStatus" SET DEFAULT 'present';
ALTER TABLE nofa.occurrence
    ALTER COLUMN "establishmentMeans" SET DEFAULT 'unknown';
```

**Comment by svoboond** [comment]
The issue with date edits has been solved (see #77), closing.
